src/test/java/com/passbook/sparkeighteen/controller/TransactionControllerTest.java

```diff
@@ -0,0 +1,122 @@
+package com.passbook.sparkeighteen.controller;
+
+import com.passbook.sparkeighteen.peristence.POJO.TransactionRequest;
+import com.passbook.sparkeighteen.peristence.POJO.TransactionResponse;
+import com.passbook.sparkeighteen.peristence.POJO.TransactionType;
+import com.passbook.sparkeighteen.service.TransactionService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith({MockitoExtension.class})
+public class TransactionControllerTest {
+    @Mock
+    private TransactionService transactionService;
+
+    @InjectMocks
+    private TransactionController transactionController;
+
+    @Test
+    public void validRequest_successfulResponse_CreditTransactionSuccessfull() throws Exception {
+
+        Integer txnID = 1;
+
+        TransactionRequest transactionRequest = TransactionRequest.builder()
+                .amount(100F)
+                .name("ketan")
+                .note("Mobile Recharge")
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        Mockito.when(transactionService.transact(txnID,transactionRequest)).thenReturn(TransactionResponse.builder()
+                .message("Credited")
+                .amount(100f)
+                .note("tea")
+                .transactionType(TransactionType.CREDIT)
+                .closingBalance(100f)
+                .build());
+
+        ResponseEntity<TransactionResponse> response = transactionController.transact(txnID,transactionRequest);
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
+
+    }
+
+    @Test
+    public void InvalidRequest_errorResponse_CreditTransactionUnSuccessfull() throws Exception {
+
+        Integer txnID = 1;
+
+        TransactionRequest transactionRequest = TransactionRequest.builder()
+                .amount(-100f)
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        Mockito.when(transactionService.transact(txnID,transactionRequest)).thenReturn(TransactionResponse.builder()
+                .message("amount cannot be less than zero")
+                .amount(null)
+                .note(null)
+                .transactionType(TransactionType.CREDIT)
+                .closingBalance(null)
+                .build());
+
+        ResponseEntity<TransactionResponse> response = transactionController.transact(txnID,transactionRequest);
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
+
+    }
+
+    @Test
+    public void validRequest_successfulResponse_DebitTransactionSuccessfull() throws Exception {
```

Comment:
naming convention ?

Reply:
It depends on individual, this was my style of naming them before which i think i am correct as i followed kunal sir.